sklearn/cluster/k_means_.py

```diff
@@ -374,6 +374,11 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
             X += X_mean
         best_centers += X_mean
 
+    if len(set(best_labels)) < n_clusters:
+        warnings.warn("number of distinct clusters found smaller than "
```

Comment:
Can you add the number of found clusters in the error messages as well as the value of n_clusters